src/Feature/InlineTitle/InlineTitleManager.ts

```diff
@@ -0,0 +1,177 @@
+                const canvas = view.canvas;
+                if (!canvas.requestFrame._originalFunc){
+                    const originalFunc = canvas.requestFrame;
+                    const manager = this;
+                    canvas.requestFrame = function() {
+                        canvas.requestFrame._originalFunc.apply(this, arguments);
+                        manager.innerUpdate(currentPath);
+                    }
+                    canvas.requestFrame._originalFunc = originalFunc;
+                }
+    
+                for (const node of canvas.nodes.values()) {
+                    promises.push(this.resolver.resolve(node.filePath).then(r => this.setCanvasTitle(node, r)));
+                }
+            }
+        }
+
+        await Promise.all(promises);
+        return promises.length > 0;
+    }
+
+    private async setCanvasTitle(node: CanvasNode, title: string | null): Promise<void> {
+        while (!node.initialized) {
+            await new Promise(resolve => setTimeout(resolve, 100));
+        }
+
+        this.logger.log(`Set canvas title "${title ?? " "}" for ${node.filePath}`);
+        this.addFakeTitleElement(node.labelEl, title);
+
+        this.addFakeTitleElement(node.placeholderEl, title);
+
+        const inlineTitleEl = node.contentEl?.querySelector(".inline-title") as HTMLElement;
+        this.addFakeTitleElement(inlineTitleEl, title);
+
+    }
+
+    private addFakeTitleElement(originalElement: HTMLElement & { fakeTitleElement?: HTMLElement}, title: string): void {
+        if (!originalElement) {
+            return;
+        }
+
+        const container = originalElement.parentElement;
+
+        if (!container) {
+            if (originalElement.fakeTitleElement) {
+                originalElement.fakeTitleElement.remove();
+                originalElement.fakeTitleElement = null;
+            }
+            return;
+        }
+
+        let el: HTMLDivElement = null;
+        for (const i of Array.from(container.children)) {
+            if (i.hasAttribute("data-ofmt") && i instanceof HTMLDivElement) {
+                el = i;
+                break;
+            }
+        }
+        if (!title) {
+            if (el) {
+                container.removeChild(el);
+            }
+            originalElement.hidden = false;
+            return;
+        }
+        if (title && el && el.innerText === title && !el.hidden) {
+            this.logger.log("skipped");
+            return;
+        }
+        if (el === null) {
+            el = document.createElement("div");
+            el.className = originalElement.className;
+            el.dataset["ofmt"] = "true";
+            el.innerText = title;
+            el.hidden = true;
+            el.onclick = () => {
+                el.hidden = true;
+                originalElement.hidden = false;
+                originalElement.focus();
+                originalElement.onmouseleave = originalElement.onblur = () => {
```

Comment:
In this case if I occasionally move mouse out of title, file will be renamed instantly.

Reply:
I think that onmouseleave is excess here.

Reply:
No, it's need for canvas label after you clicked